packages/hardhat-core/src/internal/hardhat-network/provider/types/LoggedError.ts

```diff
@@ -0,0 +1,7 @@
+export class LoggedError {
+  constructor(public readonly wrappedError: Error) {}
+}
+
+export function asLoggedError(error: Error): LoggedError {
```

Comment:
I did 2, seems more reasonable to me. I also logged the error, since we probably want to know if it happens.

Commit: f3b69cc11b44691bb272a6f7eab8190aaefa1495